src/aws_encryption_sdk/internal/formatting/encryption_context.py

```diff
@@ -47,6 +47,33 @@ def assemble_content_aad(message_id, aad_content_string, seq_num, length):
     return struct.pack(fmt, message_id, aad_content_string.value, seq_num, length)
 
 
+# help function to simplify code
+# this is necessary to pass flake8
+def _serialize_encryption_context_list(encryption_context_list, serialized_context):
+    for key, value in sorted(encryption_context_list, key=lambda x: x[0]):
+        try:
+            serialized_context.extend(
+                struct.pack(
+                    ">H{key_size}sH{value_size}s".format(key_size=len(key), value_size=len(value)),
+                    len(key),
+                    key,
+                    len(value),
+                    value,
+                )
+            )
+        # we check to make sure that we return the right type of error message for an overly long key or value
+        except struct.error as struct_error:
+            message = str(struct_error)
+            if message == "'H' format requires 0 <= number <= 65535":
```

Comment:
Minor: This is a risky way to match. If in the future the `str()` output changes even slightly, this check will no longer pass. Given that we're trying to make trapping this a useful part of our UI/UX, I would either do a check instead on a substring, add a unit test that the `"'H' format requires 0 <= number <= 65535"` message still appears when we expect it to, or both.

Reply:
did both